integrationtests/helmops/controller/controller_test.go

```diff
@@ -0,0 +1,1006 @@
+				Expect(err).ToNot(HaveOccurred())
+
+				// Bundle should be updated
+				Eventually(func(g Gomega) {
+					bundle := &fleet.Bundle{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, bundle)
+					g.Expect(err).ToNot(HaveOccurred())
+					t := []fleet.BundleTarget{
+						{
+							Name:         "default",
+							ClusterGroup: "default",
+						},
+					}
+					checkBundleIsAsExpected(g, *bundle, helmapp, t)
+
+					// make this check explicit
+					g.Expect(bundle.Spec.Helm.Chart).To(Equal("superchart"))
+				}).Should(Succeed())
+			})
+		})
+
+		When("targets is not empty", func() {
+			BeforeEach(func() {
+				targets = []fleet.BundleTarget{
+					{
+						Name:         "one",
+						ClusterGroup: "oneGroup",
+					},
+					{
+						Name:         "two",
+						ClusterGroup: "twoGroup",
+					},
+				}
+				helmapp = getRandomHelmAppWithTargets("test-not-empty", targets)
+			})
+
+			It("creates a bundle with the expected spec and the original targets", func() {
+				Eventually(func(g Gomega) {
+					bundle := &fleet.Bundle{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, bundle)
+					g.Expect(err).ToNot(HaveOccurred())
+					checkBundleIsAsExpected(g, *bundle, helmapp, targets)
+				}).Should(Succeed())
+			})
+		})
+
+		When("targets is empty and TargetCustomizations is not", func() {
+			BeforeEach(func() {
+				targets = []fleet.BundleTarget{
+					{
+						Name:         "customOne",
+						ClusterGroup: "customOneGroup",
+					},
+					{
+						Name:         "customTwo",
+						ClusterGroup: "customTwoGroup",
+					},
+				}
+				helmapp = getRandomHelmAppWithTargets("test-customizations", []fleet.BundleTarget{})
+				helmapp.Spec.TargetCustomizations = targets
+			})
+
+			It("creates a bundle with the expected spec and the customization targets", func() {
+				Eventually(func(g Gomega) {
+					bundle := &fleet.Bundle{}
+					ns := types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}
+					err := k8sClient.Get(ctx, ns, bundle)
+					g.Expect(err).ToNot(HaveOccurred())
+					checkBundleIsAsExpected(g, *bundle, helmapp, targets)
+				}).Should(Succeed())
+			})
+		})
+
+		When("targets and TargetCustomizations are not empty", func() {
+			BeforeEach(func() {
+				targets = []fleet.BundleTarget{
+					{
+						Name:         "one",
+						ClusterGroup: "oneGroup",
+					},
+					{
+						Name:         "two",
+						ClusterGroup: "twoGroup",
+					},
+				}
+				helmapp = getRandomHelmAppWithTargets("test-custom2", targets)
+				helmapp.Spec.TargetCustomizations = []fleet.BundleTarget{
+					{
+						Name:         "customOne",
+						ClusterGroup: "customOneGroup",
+					},
+					{
+						Name:         "customTwo",
+						ClusterGroup: "customTwoGroup",
+					},
+				}
+			})
+
+			It("creates a bundle with the expected spec and the targets and customization merged", func() {
```

Comment:
If we keep this behaviour, we'll want to document it. Fleet's docs website currently [states](https://fleet.rancher.io/gitrepo-targets#customization-per-cluster) that:
> The targetCustomizations: in fleet.yaml override Helm values only and do not change targeting.

Granted, that's for `fleet.yaml`, hence `GitRepo`s, but we may end up with 2 config directives named identically but which behave differently depending on the source of truth (git vs Helm).

Reply:
Customisations are gone, so this is gone as well.